src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -210,6 +204,21 @@ internal partial class Binder
                                                 continue;
                                         }
                                     }
+                                    else if (type.SpecialType != SpecialType.System_Enum)
+                                    {
+                                        if ((constraints & TypeParameterConstraintKind.ValueType) != 0)
+                                        {
+                                            // "'{0}': cannot specify both a constraint class and the 'class' or 'struct' constraint"
+                                            Error(diagnostics, ErrorCode.ERR_RefValBoundWithClass, syntax, type);
+                                            continue;
+                                        }
+                                        else if ((constraints & TypeParameterConstraintKind.ValueType) != 0)
```

Comment:
This `else if` is the same as the `if`?

Reply:
ugh, this should be checking for "unmanaged". 
I intentionally created a separate error message just for this case, but forgot to hook it up.